src/app/services/catalogue/catalogue.service.ts

```diff
@@ -37,16 +37,44 @@ export class CatalogueService {
     const topicFilter = topic ? `+topic:${topic}` : '';
     const orgFilter = org ? `org:${org}+` : '';
 
-    return this.getFromAPI(
-      `https://api.github.com/search/repositories?page=${this.CONF.page}&per_page=${this.CONF.perPage}&q=${orgFilter}fork:true${topicFilter}+sort:stars`,
-      key,
+    const perPage = this.CONF.perPage;
+    let totalSize = null;
+
+    const getPage = (page) => {
+      return this.http.get<RepositoryPage>(`https://api.github.com/search/repositories?page=${page}&per_page=${this.CONF.perPage}&q=${orgFilter}fork:true${topicFilter}+sort:stars`);
+    };
+
+    const pages$ = new Observable<Repository[]>((observer) => {
+      const emitItems = (page) => {
+        getPage(page).pipe(
+          map((res) => {
+            if (!totalSize) {
+              totalSize = res.total_count;
+            }
+            return res.items;
+          }),
+        ).subscribe((items) => {
+          observer.next(items);
+          if (perPage * page < totalSize) {
+            emitItems(page + 1);
+          } else {
+            observer.complete();
+          }
+        });
+      };
+
+      emitItems(1);
+    }).pipe(
+      reduce((acc, val) => acc.concat(val), []),
     );
```

Comment:
not sure if this is necessary

Reply:
I think it is necessary, because in the end you have multiple values in the stream. So what will happen is: on subscribe, you'll store page 1 in local storage, and displays repositories from page 1, then you'll erase page 1 with results from page 2, then display only page 2, etc...

Reply:
I see, it is necessary, but another alternative would be to get the first page to get the total size and use later forkJoin to call all the http requests at the same time and wait them.

but your solution seems quite smart by the way,

Reply:
Ah ah, I made the same comment at the same time!

Reply:
exactly haha, totally agree with you then, thanks for the great and fantastic job :rocket: